python/python/lance/torch/data.py

```diff
@@ -51,10 +51,10 @@ def _to_tensor(
             pa.types.is_floating(arr.type.value_type)
             or pa.types.is_integer(arr.type.value_type)
         ):
-            np_tensor = arr.values.to_numpy(zero_copy_only=True).reshape(
-                -1, arr.type.list_size
-            )
-            tensor = torch.from_numpy(np_tensor)
+            np_arrs = arr.to_numpy(zero_copy_only=False)
```

Comment:
cc: @eddyxu there are weird conditions that trigger this thing. I have no idea what are those

Reply:
This line significantly slow down torch training. 

@jacketsj could you investigate a bit more what data cause this line to fail?

And how much slow down it is due to this line change to pytorch kmeans?

Reply:
this `zero_copy_only=True` is part of fix to improve GPU inference by 2-3x times end-to-end, where we measured as one of the slowest step during large scale indexing.

If we can not figure out exactly what cause the failure, at least lets do:
```
try:
  tensor = torch.from_numpy(arr.values.to_numpy(zero_copy_only=True)...)
except SomeException:
  tensor = tensor.from_array(np.stack(arr.to_numpy(zero_copy_only=False)))
```

Reply:
Moved this code into #2824 after investigating.